test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -7538,13 +7538,11 @@ class XdsSecurityTest : public BasicTest {
     constexpr int kRetryCount = 100;
     int num_tries = 0;
     for (; num_tries < kRetryCount; num_tries++) {
-      // Give some time for the updates to propagate.
-      gpr_sleep_until(grpc_timeout_milliseconds_to_deadline(100));
+      // Restart the servers to force a reconnection so that previously
```

Comment:
Can you explain why this change is here?  I imagine it has something to do with the following TODO, but the exact reasoning isn't clear.

https://github.com/grpc/grpc/blob/87068e02c3c4d34f51019b3ccdceea1230c92fea/src/core/lib/security/credentials/xds/xds_credentials.cc#L167

Speaking of which, shouldn't we remove that TODO as part of this PR?

Reply:
Yes, it does :)
With this change, we were suddenly able to just reuse existing connections when there was a change in certificates. So, those existing connections were just being reused. The only time new certificates would be used are when the subchannel needs to connect again, and hence the need to restart the servers to force a reconnect